prototypes/matrix_free_stokes_global_coarsening/matrix_free_stokes_global_coarsening.cc

```diff
@@ -0,0 +1,1738 @@
+
+  FullSourceTerm<dim> source_term_function;
+
+  integrator.evaluate(EvaluationFlags::values | EvaluationFlags::gradients |
+                      EvaluationFlags::hessians);
+
+  const unsigned int cell = integrator.get_current_cell_index();
+
+  VectorizedArray<number> tau = VectorizedArray<number>(0.0);
+  std::array<number, VectorizedArray<number>::size()> h_k;
+  std::array<number, VectorizedArray<number>::size()> h;
+
+  for (auto lane = 0u; lane < matrix_free.n_active_entries_per_cell_batch(cell);
+       lane++)
+    {
+      h_k[lane] = matrix_free.get_cell_iterator(cell, lane)->measure();
+    }
+
+  for (unsigned int v = 0; v < VectorizedArray<number>::size(); ++v)
+    {
+      if (dim == 2)
+        {
+          h[v] = std::sqrt(4. * h_k[v] / M_PI);
+        }
+      else if (dim == 3)
+        {
+          h[v] = std::pow(6 * h_k[v] / M_PI, 1. / 3.);
+        }
+      tau[v] = h[v] * h[v];
+    }
+
+  for (unsigned int q = 0; q < integrator.n_q_points; ++q)
+    {
+      // Evaluate source term function
+      Point<dim, VectorizedArray<double>> point_batch =
+        integrator.quadrature_point(q);
+
+      Tensor<1, dim + 1, VectorizedArray<double>> source_value;
+
+      if (parameters.source_term == Settings::mms)
+        {
+          source_value =
+            evaluate_function<dim, double, dim + 1>(source_term_function,
+                                                    point_batch);
+        }
+
+      // Gather the original value/gradient
+      typename FECellIntegratorType::value_type value = integrator.get_value(q);
+      typename FECellIntegratorType::gradient_type gradient =
+        integrator.get_gradient(q);
+      typename FECellIntegratorType::gradient_type hessian_diagonal =
+        integrator.get_hessian_diagonal(q);
+
+      // Result value/gradient we will use
+      typename FECellIntegratorType::value_type    value_result;
+      typename FECellIntegratorType::gradient_type gradient_result;
+
+      // Assemble -nabla u + nabla p = 0 for the first 3 components
+      // The corresponding weak form is nabla v * nabla u  - p nabla \cdot v = 0
+      // ; Assemble q div(u) = 0 for the last component
+
+      for (unsigned int i = 0; i < dim; ++i)
+        {
+          gradient_result[i] = gradient[i];
+          gradient_result[i][i] += -value[dim];
+
+          value_result[dim] += gradient[i][i];
+        }
+
+      for (unsigned int i = 0; i < dim; ++i)
+        {
+          for (unsigned int k = 0; k < dim; ++k)
+            gradient_result[dim][i] += -tau * hessian_diagonal[i][k];
+        }
+      gradient_result[dim] += tau * gradient[dim];
+
+      integrator.submit_gradient(gradient_result, q);
+      integrator.submit_value(value_result, q);
+    }
+
+  integrator.integrate(EvaluationFlags::values | EvaluationFlags::gradients);
+}
+
+template <int dim, typename number>
+void
+StokesOperator<dim, number>::do_cell_integral_global(
```

Comment:
The content of this function is the same as of `do_cell_integral_local()`. Just call the above function!